lib/appsignal/hooks/sidekiq.rb

```diff
@@ -53,6 +51,7 @@ def dependencies_present?
       end
 
       def install
+        require "sidekiq/api"
```

Comment:
Wasn't sure we decided to go with this yet? It loads something that's not loaded by default and I'm not sure we'd want that.

Reply:
No decision has been made yet, but I wanted to at least have a green build to see if it worked :)

Reply:
Mike Perham's response on using this worries me still:

> note that this method was written for Web UI usage, the output could change in the future.

If we rely on this internal class our integration might break with future versions of Sidekiq if they change things around.

Reply:
If I understand correctly, we’re requiring “sidekiq/api” here specifically for the [`Sidekiq::Job#display_class`](https://github.com/mperham/sidekiq/blob/master/lib/sidekiq/api.rb#L316-L334) and [`Sidekiq::Job#sidplay_args`](https://github.com/mperham/sidekiq/blob/master/lib/sidekiq/api.rb#L336-L358) methods, which produce “human-friendly” strings.

Requiring this file makes sense, because that means we won't have to write these functions ourselves. However, they're from an internal API which could potentially change without notice in future Sidekiq versions. This could break our instrumentation, or --because the format can change without breaking our implementation-- it can change how events are grouped in AppSignal.

Another option would be to copy these methods into the project to gain absolute control over what matches, and what we send to AppSignal. We'd need to move these two functions (including their tests) to our code base, and maintain them from now on. This approach could also break on a new version of Sidekiq, but at least we're sure that the format won't change.

Reply:
Hi jeff!

Of course this is a very valid concern.
My reasoning however was that the underlying "raw data" is as much an internal matter as these methods are.

Next to that, Mike Perham also said:

> It's unlikely to change but I can't control AJ so it's hard to say.

So the question is, what is more likely to change? The methods' name/location/output, or the underlying data format? Copying the methods will prevent problems in the first case, while it will introduce new ones in case of the second.

Reply:
> Requiring this file makes sense, because that means we won't have to write these functions ourselves. However, they're from an internal API which could potentially change without notice in future Sidekiq versions. This could break our instrumentation, or --because the format can change without breaking our implementation-- it can change how events are grouped in AppSignal.

The thing with extracting it is that we only move the problem around, because when the Sidekiq argument implementation changes our own extracted methods are also broken. 

So regardless of what option we choose, if Sidekiq changes, our integration is broken. My argument for using this API is that it will be tested a lot better then our own implementation, as Sidekiq's focus is 100% on Sidekiq, while our extracted implementation is just a small detail in our gem.

Reply:
None of these options make it less likely for failure because we can’t control the input. Vendoring these methods in is a suggestion to have control over the output, at least. However, as @rmachielse said (hi Richard! :wave:), fiddling with the data is as likely to break as calling an internal method. Also, I trust Mike to maintain an internal Sidekiq method better than we could, and I trust our customers to let us know if something like this breaks. Let's ship this as is and revisit this issue later, if we need to.

Reply:
One thing we could do to prevent a catastrophic failure is to wrap the call to the sidekiq method in a begin-rescue block?

something like:

```ruby
        job_action_name =
          begin
            action_name = job.display_class
            if action_name =~ /[\.#]+/
              action_name
            else
              # Add #perform as default method name for job names without a method
              # name
              "#{action_name}#perform"
            end
          rescue NoMethodError => e
            Appsignal.logger.error("Problem parsing the Sidekiq job name: #{e}")
          end
```

Reply:
sounds like a good idea :)